cmd/gateway-azure.go

```diff
@@ -40,6 +40,19 @@ const globalAzureAPIVersion = "2016-05-31"
 const azureBlockSize = 100 * humanize.MiByte
 const metadataObjectNameTemplate = globalMinioSysTmp + "multipart/v1/%s.%x/azure.json"
 
+// Tests if s3Metadata has double "-", we cannot handle this
+// without introducing a new encoding style.
+func isValidAzureMetadata(s3Metadata map[string]string) (match bool) {
+	match = false
+	for k := range s3Metadata {
+		if strings.Contains(k, "--") {
+			match = true
+			break
+		}
+	}
+	return match
```

Comment:
The naming of the return variable can be avoided for a more concise implementation using just a single return statement in the if block and a return statement at the end of the function.

Reply:
how is that concise?

Reply:
@harshavardhana I guess below code would avoid the confusion
```go
func hasUnsupportedAzureMetadata(s3Metadata map[string]string) (matched bool) {
	for k := range s3Metadata {
		if matched = strings.Contains(k, "--"); matched {
			break
		}
	}
	return matched
}
```

Another suggestion is this function can be a closure to `s3MetaToAzureProperties()` and `s3MetaToAzureProperties()` could return error additionally.  This way we could avoid repetitive check in other places.

Reply:
> Another suggestion is this function can be a closure to s3MetaToAzureProperties() and s3MetaToAzureProperties() could return error additionally. This way we could avoid repetitive check in other places.

I wanted to avoid too many changes in this PR, but i guess we can do that.

Reply:
I think this approach would come with minimal diff i.e. within `s3MetaToAzureProperties()`